app/controllers/api/v1/peers/search_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Api::V1::Peers::SearchController < Api::BaseController
+  before_action -> { authorize_if_got_token! :read, :'read:search' }
+  before_action :set_domains
+
+  def index
+    render json: @domains
+  end
+
+  private
+
+  def set_domains
+    return if !Chewy.enabled? || params[:q].blank?
```

Comment:
Could we fallback to just a fuzzy match in PostgreSQL? Perhaps using the "matches_domain" scope?